Application/Education/ClassRegister/Diary/Frontend.php

```diff
@@ -0,0 +1,859 @@
+                        /** @var TblYear $tblYear */
+                        if ($tblYear && $tblDivision && $tblDivision->getServiceTblYear()
+                            && $tblDivision->getServiceTblYear()->getId() != $tblYear->getId()
+                        ) {
+                            continue;
+                        }
+
+                        $divisionTable[] = array(
+                            'Year' => $tblDivision->getServiceTblYear() ? $tblDivision->getServiceTblYear()->getDisplayName() : '',
+                            'Type' => $tblDivision->getTypeName(),
+                            'Division' => $tblDivision->getDisplayName(),
+                            'Option' => new Standard(
+                                '',  self::BASE_ROUTE . '/Selected', new Select(),
+                                array(
+                                    'DivisionId' => $tblDivision->getId(),
+                                    'BasicRoute' => self::BASE_ROUTE . '/Teacher'
+                                ),
+                                'Auswählen'
+                            )
+                        );
+                    }
+                }
+
+                if (empty($divisionTable)) {
+                    $table = new Warning('Das pädagogisches Tagebuch steht 
+                        nur Klassenlehrern und Tutoren zur Verfügung.', new Exclamation());
+                } else {
+                    $table = new TableData($divisionTable, null, array(
+                        'Year' => 'Schuljahr',
+                        'Type' => 'Schulart',
+                        'Division' => 'Klasse',
+                        'Option' => ''
+                    ), array(
+                        'order' => array(
+                            array('0', 'desc'),
+                            array('1', 'asc'),
+                            array('2', 'asc'),
+                        ),
+                        'columnDefs' => array(
+                            array('type' => 'natural', 'targets' => 2)
+                        ),
+                    ));
+                }
+            }
+        }
+
+        $Stage->setContent(
+            new Layout(array(
+                new LayoutGroup(array(
+                    new LayoutRow(array(
+                        empty($buttonList)
+                            ? null
+                            : new LayoutColumn($buttonList),
+                        $table
+                            ? new LayoutColumn(array($table))
+                            : null
+                    ))
+                ), new Title(new Select() . ' Auswahl'))
+            ))
+        );
+
+        return $Stage;
+    }
+
+    /**
+     * @param bool $IsAllYears
+     * @param bool $IsGroup
+     * @param null $YearId
+     *
+     * @return Stage
+     */
+    public function frontendHeadmasterSelectDivision($IsAllYears = false, $IsGroup = false, $YearId = null)
```

Comment:
Codacy found an issue: [The method frontendHeadmasterSelectDivision() has a Cyclomatic Complexity of 13. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)